homeassistant/components/braviatv/__init__.py

```diff
@@ -44,3 +67,223 @@ async def async_unload_entry(hass, config_entry):
+            await self.hass.async_add_executor_job(
+                self.braviarc.volume_up, self.audio_output
+            )
+            await self.async_request_refresh()
+
+    async def async_volume_down(self):
+        """Send volume down command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(
+                self.braviarc.volume_down, self.audio_output
+            )
+            await self.async_request_refresh()
+
+    async def async_volume_mute(self, mute):
+        """Send mute command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.mute_volume, mute)
+            await self.async_request_refresh()
+
+    async def async_media_play(self):
+        """Send play command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_play)
+            self.playing = True
+            await self.async_request_refresh()
+
+    async def async_media_pause(self):
+        """Send pause command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_pause)
+            self.playing = False
+            await self.async_request_refresh()
+
+    async def async_media_stop(self):
+        """Send stop command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_stop)
+            self.playing = False
+            await self.async_request_refresh()
+
+    async def async_media_play_pause(self):
+        """Simulate play/pause command."""
+        if self.playing:
+            await self.async_media_pause()
+        else:
+            await self.async_media_play()
+
+    async def async_media_next_track(self):
+        """Send next track command."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_next_track)
+            await self.async_request_refresh()
+
+    async def async_media_previous_track(self):
+        """Send previous track command."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_previous_track)
+            await self.async_request_refresh()
+
+    async def async_select_source(self, source):
+        """Set the input source."""
+        if source in self.content_mapping:
+            uri = self.content_mapping[source]
+            async with self.state_lock:
+                await self.hass.async_add_executor_job(self.braviarc.play_content, uri)
+                await self.async_request_refresh()
+
+    async def async_send_command(self, command, repeats=1):
+        """Send command to device."""
+        async with self.state_lock:
+            for _ in range(repeats):
+                for cmd in command:
+                    await self.hass.async_add_executor_job(
```

Comment:
It would be better to make a function that loops and does the sync calls and then schedule that function once on the executor.

Reply:
done